### PR TITLE
[Hyper-V extension] Use OOP WinGet object. Update or install Desktop App Installer package during Hyper-V Configure workflow if needed.

### PR DESCRIPTION
**Pull request by sshilov7**
Replace usage of in-proc WinGet object with OOP one in Hyper-V configure workflow.
Add check for version of Microsoft.DesktopAppInstaller package installed on the target Hyper-V VM. If the version is less than 1.22.10661.0 update it to the latest in MS Store using Store APIs.

## Validation steps performed
Tested OOP WinGet and install and update of Microsoft.DesktopAppInstaller on 19041.

## PR checklist
- [ ] Closes #xxx
- [ ] Tests added/passed
- [ ] Documentation updated

### CODE REVIEW

File: HyperVExtension/src/DevSetupEngine/ConfigurationFileHelper.cs [hunk 1]
```diff
@@ -44,34 +47,130 @@ public ConfigurationFileHelper()
...
+
+        AppInstallManager installManager = new();
+        var progressWatcher = new ApplyConfigurationProgressWatcher(progress);
+        var configurationSetChangeData = GetConfigurationSetChangeData(AppInstallerPackageName, DevSetupEngineTypes.ConfigurationUnitState.Pending, string.Empty);
+        progressWatcher.Report(configurationSetChangeData);
+
+        AppInstallItem? installItem;
+        if (doInstall)
+        {
+            installItem = await installManager.StartAppInstallAsync(AppInstallerStoreId, null, true, false);
```
Comment (huzaifa-d) [inline HyperVExtension/src/DevSetupEngine/ConfigurationFileHelper.cs:79]: Does this need to be under try/catch?

File: HyperVExtension/src/DevSetupEngine/ConfigurationFileHelper.cs [hunk 2]
```diff
@@ -44,34 +47,130 @@ public ConfigurationFileHelper()
...
+                var description = GetInstallStatusDescription(installStatus);
+                configurationSetChangeData = GetConfigurationSetChangeData(AppInstallerPackageName, DevSetupEngineTypes.ConfigurationUnitState.InProgress, description);
+                progressWatcher.Report(configurationSetChangeData);
+                await Task.Delay(5000);
+                installStatus = installItem.GetCurrentStatus();
+            }
+#else
+            try
+            {
+                await Task.Delay((int)TimeSpan.FromMinutes(15).TotalMilliseconds, cancellationToken.Token);
```
Comment (huzaifa-d) [inline HyperVExtension/src/DevSetupEngine/ConfigurationFileHelper.cs:131]: Nit: await Task.Delay(TimeSpan.FromMinutes(15), cancellation.Token); will also work.

**Review by huzaifa-d — APPROVED** [review]

File: HyperVExtension/src/DevSetupEngine/ConfigurationFileHelper.cs (hunk 1 above)
Comment (sshilov7) [reply HyperVExtension/src/DevSetupEngine/ConfigurationFileHelper.cs:79]: No, this method intentionally doesn't have try/catch. The caller has it and will convert exceptions into failure response and complete the async operation.

File: HyperVExtension/src/DevSetupEngine/ConfigurationFileHelper.cs [hunk 3]
```diff
@@ -1,12 +1,13 @@
...
 
-using System.Buffers;
+using System;
 using Microsoft.Management.Configuration;
-using Microsoft.Management.Configuration.Processor;
-using Microsoft.Management.Configuration.SetProcessorFactory;
+using Windows.ApplicationModel;
+using Windows.ApplicationModel.Store.Preview.InstallControl;
+using Windows.Foundation;
+using Windows.Media.Playback;
```
Comment (bbonaby) [inline HyperVExtension/src/DevSetupEngine/ConfigurationFileHelper.cs:9]: I think this one can be removed

**Review by bbonaby — APPROVED** [review]

**Pull request merged**